pkg/solana/logpoller/log_poller.go

```diff
@@ -2,40 +2,62 @@ package logpoller
 
 import (
 	"context"
+	"encoding/base64"
 	"errors"
+	"fmt"
+	"math"
 	"time"
 
+	"github.com/lib/pq"
 	"github.com/smartcontractkit/chainlink-common/pkg/logger"
 	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/client"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/internal"
 )
 
 var (
 	ErrFilterNameConflict = errors.New("filter with such name already exists")
 )
 
 type ORM interface {
+	ChainID() string
 	InsertFilter(ctx context.Context, filter Filter) (id int64, err error)
 	SelectFilters(ctx context.Context) ([]Filter, error)
 	DeleteFilters(ctx context.Context, filters map[int64]Filter) error
 	MarkFilterDeleted(ctx context.Context, id int64) (err error)
 	MarkFilterBackfilled(ctx context.Context, id int64) (err error)
+	InsertLogs(context.Context, []Log) (err error)
+	SelectSeqNums(ctx context.Context) (map[int64]int64, error)
+}
+
+type ILogPoller interface {
+	Start(context.Context) error
+	Close() error
+	RegisterFilter(ctx context.Context, filter Filter) error
+	UnregisterFilter(ctx context.Context, name string) error
+	Process(programEvent ProgramEvent) error
 }
 
 type LogPoller struct {
 	services.Service
 	eng *services.Engine
+	services.StateMachine
 
-	lggr logger.SugaredLogger
-	orm  ORM
+	lggr      logger.SugaredLogger
+	orm       ORM
+	client    internal.Loader[client.Reader]
+	collector *EncodedLogCollector
 
 	filters *filters
 }
 
-func New(lggr logger.SugaredLogger, orm ORM) *LogPoller {
+func New(lggr logger.SugaredLogger, orm ORM, cl internal.Loader[client.Reader]) ILogPoller {
```

Comment:
Using `New` here is great and helps provide a better coding experience. You can use `logpoller.New` instead of `logpoller.NewLogPoller` which would be repetative. However, we really should not be returning an interface here. Instead we should return a pointer type, which is Go convention.